### DISCUSSION TITLE
401 Unauthorized response to REST API requests

### DISCUSSION THREAD

**Discussion by JoachimSand** [iTwin/itwinjs-core#4087] [Q&A] [answered]
Hi,

I suspect this issue might be related to #3011, but since my issue pertains to a native application I thought a separate discussion was needed.

I have recently been developing a native application in Unity which visualizes data from iTwin. 
To access issues, projects and other relevant data we have recently started incorporating the iTwin REST API into our workflow. Initially, this was working without any problems. However, today all my requests have been returning ```401 Unauthorized``` with the message ``` "Access denied due to invalid subscription key. Make sure to provide a valid key for an active subscription."```. 

![image](https://user-images.githubusercontent.com/37040245/184643497-9a6dc556-7aaf-4a3e-abef-ae71e4152c9c.png)

I have registered a native application with all available scopes:
``insights:read validation:read mesh-export:read imodelaccess:read itwinjs organization profile projects:read users:read contextcapture:read insights:modify webhooks:modify webhooks:read forms:modify changedelements:read forms:read issues:modify issues:read designelementclassification:modify changedelements:modify designelementclassification:read export:read transformations:read synchronization:read mesh-export:modify realitydataanalysis:read openid contextcapture:modify clashdetection:read validation:modify projects:modify synchronization:modify imodels:read library:read realitydataanalysis:modify realitydata:read savedviews:read realitydata:modify storage:read storage:modify imodels:modify savedviews:modify transformations:modify email export:modify library:modify clashdetection:modify``

These scopes are all present in my client as well. I obtain access tokens via the IModelHost class. I have tried printing these tokens out and doing the same requests using curl, but these result in the same ```401 Unauthorized``` error. However, if I use the "Test it out" feature on e.g. https://developer.bentley.com/apis/issues-v1/operations/get-project-issues/ and generate the access token from there the request is completed properly.

Just to reiterate: This setup was working without any issues on the 12th of August, but has now broken on the 15th of August.

**Comment by mgooding** [comment]
Thank you for submitting this question. Off the top of my head I can't think of why this should've stopped working for you. I've reached out to our internal support for iTwin Platform APIs and will let you know what I hear.

**Reply by mgooding** [reply]
The team is looking into this, but may not be able to respond until tomorrow due to a combination of public holidays and conference travel.

**Reply by HaleRazor73** [reply]
@JoachimSand , could you confirm that you are still having this issue and that you are still on the trial subscription tier?  As @mgooding mentioned, most of our team is on holiday today and I am in a conference.  My ability to communicate and to troubleshoot will be limited today.  I will do my best to put time into this during my conference breaks.

**Reply by JoachimSand** [reply]
@KevinHale-Bentley Yes, I am still experiencing this issue. As far as I know my organization is indeed on a trial subscription, but will be moving to a full license once the trial has expired.

**Reply by SammyMig** [reply]
@JoachimSand we have identified an issue and we are urgently working to resolve it.

**Reply by arnab-git-hub** [reply]
@JoachimSand Our internal tests indicate that you may have created a client for the iTwin Platform that is experiencing errors accessing our APIs.  We’re working urgently to correct this issue.  In the interim, if you need immediate access to the iTwin APIs you should be able to go to https://developer.bentley.com/ and recreate our client and used the new client ID to access our APIs.  We apologize for any inconvenience and ask that if you have any questions you contact our support group directly.

**Comment by JoachimSand** [comment, marked as answer]
The issue has now been resolved internally by Bentley. My client is now working exactly as before. Thank you for the assistance @ArnabGogoi @KevinHale-Bentley @SammyMig @mgooding.
